contrib/gorilla/mux/mux_test.go

```diff
@@ -11,70 +13,76 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestHttpTracer200(t *testing.T) {
-	assert := assert.New(t)
-	mt := mocktracer.Start()
-	defer mt.Stop()
+var httpTests = []struct {
+	responseCode int
+	httpMethod   string
+	url          string
+}{
+	{http.StatusOK, "GET", "/200"},
+	{http.StatusNotFound, "GET", "/not_a_real_route"},
+	{http.StatusMethodNotAllowed, "POST", "/405"},
+	{http.StatusInternalServerError, "GET", "/500"},
+}
 
-	// Send and verify a 200 request
-	url := "/200"
-	r := httptest.NewRequest("GET", url, nil)
-	w := httptest.NewRecorder()
-	router().ServeHTTP(w, r)
-	assert.Equal(200, w.Code)
-	assert.Equal("OK\n", w.Body.String())
+func TestHttpTracer(t *testing.T) {
+	for _, ht := range httpTests {
+		t.Run(ht.httpMethod+ht.url, func(t *testing.T) {
+			assert := assert.New(t)
+			mt := mocktracer.Start()
+			defer mt.Stop()
+			responseCodeStr := strconv.Itoa(ht.responseCode)
 
-	// Ensure the request is properly traced
-	spans := mt.FinishedSpans()
-	assert.Equal(1, len(spans))
+			// Send and verify a request
+			r := httptest.NewRequest(ht.httpMethod, ht.url, nil)
+			w := httptest.NewRecorder()
+			router().ServeHTTP(w, r)
+			assert.Equal(ht.responseCode, w.Code)
+			assert.Equal(responseCodeStr+"!\n", w.Body.String())
 
-	s := spans[0]
-	assert.Equal("http.request", s.OperationName())
-	assert.Equal("my-service", s.Tag(ext.ServiceName))
-	assert.Equal("GET "+url, s.Tag(ext.ResourceName))
-	assert.Equal("200", s.Tag(ext.HTTPCode))
-	assert.Equal("GET", s.Tag(ext.HTTPMethod))
-	assert.Equal(url, s.Tag(ext.HTTPURL))
-	assert.Equal(nil, s.Tag(ext.Error))
-}
+			spans := mt.FinishedSpans()
+			assert.Equal(1, len(spans))
 
-func TestHttpTracer500(t *testing.T) {
-	assert := assert.New(t)
-	mt := mocktracer.Start()
-	defer mt.Stop()
+			s := spans[0]
+			assert.Equal("http.request", s.OperationName())
+			assert.Equal("my-service", s.Tag(ext.ServiceName))
+			assert.Equal(responseCodeStr, s.Tag(ext.HTTPCode))
+			assert.Equal(ht.httpMethod, s.Tag(ext.HTTPMethod))
+			assert.Equal(ht.url, s.Tag(ext.HTTPURL))
 
-	// Send and verify a 500 request
-	url := "/500"
-	r := httptest.NewRequest("GET", url, nil)
-	w := httptest.NewRecorder()
-	router().ServeHTTP(w, r)
-	assert.Equal(500, w.Code)
-	assert.Equal("500!\n", w.Body.String())
+			// Response code dependant tests
+			switch ht.responseCode {
```

Comment:
This defeats the purpose of having table driven tests. Let's drop the switch-case, and instead include expected resource name and error in the test structure.